engine/grid.py

```diff
@@ -319,12 +319,13 @@ def find_border_nodes(self):
     
     ##Return bottom most node that is activated in the right column
     def bottom_rightmost_node(self, pos):
-        candidate_nodes = [node for node in self.border_nodes if node.y == pos[1]+1]
+        # node_info: (node, row, col)
+        candidate_nodes = [node_info for node_info in self.border_nodes if node_info[2] == pos[1]+1]
         if (candidate_nodes == []):
             return None
         else:
-            node = min(candidate_nodes,key=lambda node: node.x)
-            return (node.x,node.y)
+            node_info = min(candidate_nodes,key=lambda node_info: node_info[1])
+            return (node_info[1],node_info[2])
```

Comment:
@AdityaKompella6 This (and the respective adjustments in find_border_nodes when making border_list) is where we found the "bug" Lin mentioned in the PR description. Could you double-check this to make sure this makes sense with you?

So, rather than using a nodes x,y values (as it may not be a perfect integer, but rather a float value as we are creating a grid from lat,long values) we changed it to use row, column values as we realized the test_activation_shapes script was set up to initialize nodes x,y's using row, column.

Reply:
@jinlin270 I think I get the row, col -> x/y-axis relationship in this function (or maybe this is a 3am brain fart)

node_info[1] corresponds to the row this node is located (aka y-value). 
node_info[2] corresponds to the column this node is located (aka x-value).

I think here and in `get_all_lawnmower_waypoints_adjustable` we are traversing **vertically** (rather than horizontally like we discussed in full team).

So, `node_info[2] == pos[1]+1` searches for all the nodes in the column to the right of the current column. Then, `node_info = min(candidate_nodes,key=lambda node_info: node_info[1])` finds the node out of those with the lowest row. So, this new bottom_rightmost_node will be used in 'get_all_lawnmower_waypoints_adjustable', which iterates over this node and all the ones above it in the same column.

@AdityaKompella6 if you could confirm this that would be amazing 🛐

Reply:
Yep float to int makes sense. Also starting at the bottom right is correct.